core/domain/exp_domain.py

```diff
@@ -958,6 +967,20 @@ def _verify_no_dead_ends(self):
                 'It is impossible to complete the exploration from the '
                 'following states: %s' % ', '.join(dead_end_states))
 
+    def validate_exploration_matches_content(
```

Comment:
Consider just putting this in the services layer. Use the domain object to retrieve the content HTML and compare it in services. Then you don't need these two extra methods.